app/content/webviewPreload.js

```diff
@@ -2,10 +2,29 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var ipc = require('electron').ipcRenderer
 var messages = require('../../js/constants/messages')
 var KeyCodes = require('../../js/constants/keyCodes')
 
+var ipc = {
+  events: {},
+  embedder: null,
+  processMessage: function (event) {
+    if (event.origin === 'file://') {
```

Comment:
It only sends file:// unfortunately. The updates I'm making now to Electron will fire a "HELO" message to register the embedder as soon as the script context is created and before the page will have any opportunity to preempt it.